plugins/modules/ec2_instance.py

```diff
@@ -1895,7 +2018,9 @@ def main():
                 msg='No matching instances found',
                 changed=False,
             )
-    elif existing_matches:
+    elif module.params.get('exact_count'):
+        enforce_count(module)
```

Comment:
```suggestion
        enforce_count(module)
```

If you pass in existing_matches here, there's no need for an additional search.  filters already supports filtering by tag.